projects/components/src/table/table.component.ts

```diff
@@ -330,6 +343,85 @@ export class TableComponent
     this.rowStateSubject.complete();
     this.columnStateSubject.complete();
     this.columnConfigsSubject.complete();
+
+    this.mouseMoveFunction && this.mouseMoveFunction();
+    this.mouseUpFunction && this.mouseUpFunction();
+  }
+
+  public trackItem(_index: number, column: TableColumnConfigExtended): string {
+    return `${column.id}-${column.width}`;
+  }
+
+  public queryHeaderRowElement(): HTMLElement {
+    return this.elementRef.nativeElement.querySelector('cdk-header-row');
+  }
+
+  public queryHeaderCellElement(index: number): HTMLElement {
+    return this.elementRef.nativeElement.querySelectorAll('cdk-header-cell').item(index);
+  }
+
+  private getVisibleColumnConfig(index: number): TableColumnConfigExtended {
+    return this.columnConfigsSubject.value.filter(cc => cc.visible)[index];
+  }
+
+  public onResizeMouseDown(event: MouseEvent, index: number): void {
+    const leftColumn: ColumnInfo = this.buildColumnInfo(index - 1);
+    const rightColumn: ColumnInfo = this.buildColumnInfo(index);
+
+    this.lastOffsetX = this.calcOffsetX(event, 0, leftColumn.bounds.left, rightColumn.bounds.right);
+    this.mouseMoveFunction = this.renderer.listen('document', 'mousemove', e =>
```

Comment:
Why do we need these on the document rather than on the element? Offsets? It would seem those offsets should be able to be local to the component.

Reply:
Are these being removed? Also are they being invoked on events outside of table?